pyrealm/phenology/fapar_limitation.py

```diff
@@ -0,0 +1,208 @@
+"""Class to compute the fAPAR_max and annual peak Leaf Area Index (LAI)."""
+
+import numpy as np
+from numpy.ma.core import zeros
+from numpy.typing import NDArray
+from typing_extensions import Self
+
+from pyrealm.pmodel import PModel, SubdailyScaler
+
+
+def get_annual(
+    x: NDArray,
+    datetimes: NDArray[np.datetime64],
+    growing_season: NDArray[np.bool],
+    method: str,
+) -> NDArray:
+    """Computes an array of the annual total or mean of an entity x given datetimes.
+
+    Args:
+        x: Array of values to be converted to annual values. Should be either daily (
+        same datetimes as growing_season) or subdaily (same datetimes as datetimes
+        array)
+        datetimes: Datetimes of the measurements as np.datetime62 arrays.
+        growing_season: Bool array of days, indicating whether they are ain growing
+        season or not.
+        method: Either "total" (sum all values of the year) or "mean" (take the mean
+        of all values of the year)
+    """
+
+    # Extract years from datetimes
+    all_years = [np.datetime64(i, "Y") for i in datetimes]
+
+    # Create scaler object to handle conversion between scales
+    scaler = SubdailyScaler(datetimes)
+    scaler.set_nearest(np.timedelta64(12, "h"))
```

Comment:
We can't use the `SubdailyScalar` on data that isn't sampled at subdaily frequency. The code explicitly checks for the spacing of observations within a day and breaks with coarser data (daily, monthly). I think we can still use the same approach if we nest this within the `if` statements below and solve it for the daily+ mode in another way.

Reply:
We should add a failure mode to SubdailyScalar for that case!

Reply:
I opened issue #416 for this.